### PR TITLE
Feat/AUT-345/messages for choices constraints

### PR DESCRIPTION
**Pull request by bziondik**
Related to: https://oat-sa.atlassian.net/browse/AUT-345
image

**Acceptance criteria**
GIVEN a choiceInteraction with N defined number of choices:
AC2: Response properties

GIVEN AC1a
WHEN user navigates to Response tab of the choiceInteraction widget
THEN TAO, based on the predefined or custom options selected for the choice constraints, shall present the following messages:

Single choice:
    1. Constraint: None → Please define the correct response below.
    2. Constraint: Answer required → Please define the correct response below.

Multiple Choice:
    3. Constraint: None → Please define the correct response below.
    4. Constraint: Answer required → Please define the correct response below. / You need to select at least 1 choice.
    5. Constraint: Other constraints →
         a. minChoices ≠ Disabled / maxChoices ≠ Disabled -> Please define the correct response below. / You need to select from    {minChoices value} to {maxChoices value} choices.
         b. minChoices ≠ Disabled / maxChoices = Disabled -> Please define the correct response below. / You need to select at least {minChoices value} choices.
         c. minChoices = Disabled / maxChoices ≠ Disabled -> Please define the correct response below. / You can select up to {maxChoices value} choices.
         d. minChoices = maxChoices -> Please define the correct response below. / You need to select {minChoices = maxChoices value} choices.

### CODE REVIEW

**Review by marpesia — CHANGES_REQUESTED** [review]
Great job! 
I only find one scenario that is not working as expected. Please check it :)

File: src/qtiCommonRenderer/renderers/interactions/ChoiceInteraction.js [hunk 1]
```diff
@@ -209,73 +202,96 @@ var _setInstructions = function _setInstructions(interaction) {
...
+        // Multiple Choice: 5. Constraint: Other constraints -> minChoices ≠ Disabled / maxChoices ≠ Disabled -> “You need to select {minChoices = maxChoices value} choices.“
+        msg = __('You need to select %s choices', min);
+        instructionMgr.appendInstruction(interaction, msg, function () {
+            if (_getRawResponse(interaction).length === min) {
+                this.setLevel('success');
+            } else {
+                this.reset();
+            }
+        });
+    } else if (max > 1 && typeof min === 'undefined') {
```
[the remark this answers is not in the document]
Comment (bziondik) [reply src/qtiCommonRenderer/renderers/interactions/ChoiceInteraction.js]: fixed

**Review by marpesia — APPROVED** [review]
**Review Checklist**
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [ ] Commits are atomic
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by Silvia-loza — APPROVED** [review]
All looks good and works as expected :clap:
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X] New code is respecting code style rules
- [X] New code is respecting best practices
- [X]  Pull request's target is not` master`
- [X] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Pull request merged**